src/sentry/api/serializers/models/organization_member.py

```diff
@@ -64,20 +74,27 @@ def get_attrs(
         external_users_map = defaultdict(list)
 
         if "externalUsers" in self.expand:
-            external_users = list(ExternalUser.objects.filter(organizationmember__in=item_list))
-
-            for external_user in external_users:
-                serialized = serialize(external_user, user)
-                external_users_map[external_user.organizationmember_id].append(serialized)
+            organization_id = get_organization_id(item_list)
+            external_actors = list(
+                ExternalActor.objects.filter(
+                    actor_id__in={user.actor_id for user in users_set},
+                    organization_id=organization_id,
+                )
+            )
+
+            serialized_list = serialize(external_actors, user, key="user")
+            for serialized in serialized_list:
+                external_users_map[serialized["userId"]].append(serialized)
```

Comment:
The id here might be cast to a string, we often do this in our serializers. Probably worth checking, since we try to look up via an int later on